frontend/src/wrapper/TopNavigation.tsx

```diff
@@ -111,81 +112,115 @@ export default function TopNavigation({
+                  <ListItemIcon>
+                    <DarkModeIcon fontSize='small' />
+                  </ListItemIcon>
+                  <Switch
+                    size='small'
+                    checked={localStorage.getItem('dark_mode_enabled') === 'true'}
+                    onChange={toggleDarkMode}
+                    inputProps={{ 'aria-label': 'controlled' }}
+                  />
+                </MenuItem>
+              </span>
+            </Tooltip>
+            <Link href='/api/logout' color='inherit' underline='none'>
+              <MenuItem data-test='logoutLink'>
+                <ListItemIcon>
+                  <LogoutIcon fontSize='small' />
+                </ListItemIcon>
+                <ListItemText>Sign Out</ListItemText>
+              </MenuItem>
+            </Link>
+          </Menu>
+        </Box>
+        <Box sx={{ flexGrow: 1, ml: 2, display: { cursor: 'pointer' } }}>
+          <Link href='/beta' color='inherit' underline='none' style={{ color: 'inherit', textDecoration: 'inherit' }}>
+            <Typography variant='h5' component='div'>
+              <span style={{ fontFamily: 'Pacifico' }}>Bailo</span>
+              {betaAdornment}
+            </Typography>
+          </Link>
         </Box>
-        <Stack direction='row' spacing={2} justifyContent='center' alignItems='center'>
-          <ExpandableButton
-            label='Add Model'
-            icon={<Add />}
-            onClick={() => handleNewModelClicked()}
-            ariaLabel='Add a new model'
-          />
-          {currentUser ? (
-            <>
-              <IconButton onClick={handleUserMenuClicked} data-test='userMenuButton'>
-                <UserAvatar entity={{ kind: EntityKind.USER, id: currentUser.dn }} size='chip' />
-              </IconButton>
-              <Menu sx={{ mt: '10px', right: 0 }} anchorEl={anchorEl} open={actionOpen} onClose={handleMenuClose}>
-                <MenuList>
-                  {/* TODO - currently breaks v1. Re-add when v2 is fully adopted */}
-                  <Tooltip title='This feature has been temporarily disabled'>
-                    <span>
-                      <MenuItem disabled data-test='toggleDarkMode'>
+        <Box sx={{ display: { xs: 'none', sm: 'flex' } }}>
+          <Stack direction='row' spacing={2} justifyContent='center' alignItems='center'>
+            <ExpandableButton
+              label='Add Model'
+              icon={<Add />}
+              onClick={() => handleNewModelClicked()}
+              ariaLabel='Add a new model'
+              height='40px'
+            />
+            <ModelSearchField />
+            {currentUser ? (
+              <>
+                <IconButton onClick={handleUserMenuClicked} data-test='userMenuButton'>
+                  <UserAvatar entity={{ kind: EntityKind.USER, id: currentUser.dn }} size='chip' />
+                </IconButton>
+                <Menu
+                  sx={{ mt: '10px', right: 0 }}
```

Comment:
Not your doing, but we shouldn't be using `right: 0` to position this element. 

We should also be using `theme.spacing` to set margin/padding.

Up to you if you want to address it here though.